### PR TITLE
Enable skipped E2E tests for attributes

### LINKED ISSUE #50027: [E2E test]: Fix the E2E tests for product attributes
### Describe the solution you'd like

Fix the E2E test for product attributes in the blocks product editor.

The tests to check that it's possible to add existing attributes [have been marked as flaky](https://github.com/woocommerce/woocommerce/issues/48883). They are being skipped ([create product](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js#L270) and [update attributes](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js#L360)) now, but we should fix them.

### PR DESCRIPTION
**Pull request by octaedro**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds a few changes to reenable two skipped tests:
- [create product](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js#L270)
- [update attributes](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js#L360)

Closes #50027.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

### Running E2E tests locally

```
cd plugins/woocommerce
```

Ensure Playwright is installed

```
npx playwright install
```

Ensure wp-env is running

```
pnpm env:start
```

Run the following tests

```cli
pnpm test:e2e tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js
```

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js [hunk 1]
```diff
@@ -267,36 +267,31 @@ test(
...
-			await getAttributesResponsePromise;
 		} );
 
 		await test.step( 'add an existing attribute', async () => {
 			await page.getByRole( 'button', { name: 'Add new' } ).click();
 
 			await page.waitForLoadState( 'domcontentloaded' );
 
-			// Add attributes that do not exist
-			await page.getByPlaceholder( 'Search or create attribute' ).click();
```
Comment (octaedro) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js:293]: This placeholder doesn't exist anymore.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js [hunk 2]
```diff
@@ -355,9 +350,7 @@ test.skip(
 	}
 );
 
-// Test skipped because an issue with the options not always loading makes it flaky.
-// See https://github.com/woocommerce/woocommerce/issues/44925
```
Comment (louwie17) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js:359]: Did you try to reproduce this issue? Or look at the code to see why that could be happening?
I know this is also a place where we did want to use Damian's implementation of the form token field, so maybe its worth replacing that first?

**Review by louwie17 — COMMENTED** [review]
Just left a comment around a deleted comment.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js (hunk 2 above)
Comment (octaedro) [reply plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js:359]: Hi @louwie17,

> Did you try to reproduce this issue? Or look at the code to see why that could be happening?

Yes, of course. We were trying to get the button by a label that was not present on the screen. That made this test fail consistently. I couldn't make the skipped test work even once on my end, so they seem like a broken test, more than a flaky one.
I changed the way we were getting the remove button (X) to use the `aria-label`. That fixed the problem here.

As was mentioned [in this comment](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js#L373), we reload and retry to get the element to address the same problem. 

> I know this is also a place where we did want to use Damian's implementation of the form token field, so maybe its worth replacing that first?

That seems out of scope. I would prefer to fix this test now and tackle the improvement in a follow-up PR in the future. After having Damian's implementation in place, I would modify this test as part of that PR.

**Review by louwie17 — APPROVED** [review]
Tested well and changes look good, thanks @octaedro !

**Pull request merged**